src/extension/tab/tab.ts

```diff
@@ -2,25 +2,33 @@
 import "./tabRelay";
 import browser from "webextension-polyfill";
 
-/* 
-  Content scripts are unable to modify the window object directly. 
-  A common workaround for this issue is to inject an inlined function
-  into the inspected tab.
-*/
-if (typeof document === "object" && document instanceof HTMLDocument) {
+injectScriptSync(browser.runtime.getURL("hook.js"));
+
+// In this content script we have access to DOM, but don't have access to the webpage's window,
+// so we inject this inline script tag into the webpage (allowed in Manifest V2).
+// In Manifest V3, we'll have to switch this approach for Chrome-based browsers.
+// This function is based on
+// https://github.com/facebook/react/blob/18a9dd1c60fdb711982f32ce5d91acfe8f158fe1/packages/react-devtools-extensions/src/contentScripts/prepareInjection.js
+// which is released under a MIT license (Copyright (c) Meta Platforms, Inc. and affiliates.) that can be found here:
+// https://github.com/facebook/react/blob/18a9dd1c60fdb711982f32ce5d91acfe8f158fe1/LICENSE
+function injectScriptSync(src: string) {
+  let code = "";
+  const request = new XMLHttpRequest();
+  request.addEventListener("load", function () {
+    code = this.responseText;
+  });
+  request.open("GET", src, false);
+  request.send();
+
   const script = document.createElement("script");
-  script.setAttribute("type", "module");
-  script.setAttribute("src", browser.runtime.getURL("hook.js"));
-  document.addEventListener("DOMContentLoaded", () => {
-    const importMap = document.querySelector('script[type="importmap"]');
-    if (importMap != null) {
-      importMap.parentNode?.insertBefore(script, importMap.nextSibling);
-    } else {
-      const head =
-        document.head ||
-        document.getElementsByTagName("head")[0] ||
-        document.documentElement;
-      head.insertBefore(script, head.lastChild);
+  script.textContent = code;
```

Comment:
By retrieving the text and using `textContent`, we can synchronously insert and remove the injected `script` tag with no change of any hydration being confused by it.